remove the hardcoded route to frontpage and to search in all editions

davidgrove73 opened the pull request:
This fixes the routing problems when searching in all editioins

Should propably be testes by someone else than me

We can also simplify some of the code, but we can leave that for later.

ZahraMousavi commented:
If you make an empty search. Then change the view and then change the language. It omits the subject name and gives an error.

davidgrove73 commented:
I have now fixed the problems with the change language url
and i have removed the page param from urls in the navigation tree and from the remove constraint url
